rosbag2_transport/src/rosbag2_transport/player.cpp

```diff
@@ -222,6 +231,22 @@ void Player::prepare_clock(const PlayOptions & options, rcutils_time_point_value
 {
   double rate = options.rate > 0.0 ? options.rate : 1.0;
   clock_ = std::make_unique<rosbag2_cpp::TimeControllerClock>(starting_time, rate);
+
+  // Create /clock publisher
+  if (options.clock_publish_frequency > 0.f) {
```

Comment:
It looks like that  `Player::prepare_publishers(const PlayOptions & options)` would be a more appropriate place for creating publisher for the `/clock` topic. \
It least I would expect to search for it there if I would try elaborate with source code.

Reply:
I disagree - this is a pretty different feature. It doesn't help that there aren't docstrings on the functions, but `prepare_publishers` is, in my mind, explicitly for the playback publishers, which have metadata from the storage.

This is probably a matter of taste, I think I'm going to not take this note.

Reply:
@emersonknapp Can we come to some compromise solution? \
I would suggest to move code related to the creation of the publisher for the `/clock` topic to the separate function. \
May be name it something like `prepare_clock_publisher(clock_publish_frequency)`.

It's kind of separate operation and doesn't really directly relates to the `Player::prepare_clock(..)` to be inside of it.

Reply:
Since this is a matter of taste, I am going to move forward as-is - feel free to open a follow-up PR afterwards if it feels important.